src/sparse/KokkosSparse_CrsMatrix.hpp

```diff
@@ -494,14 +494,36 @@ class CrsMatrix {
     //as the constructor of StaticCrsGraph does not allow copy from non const version.
   }
 
+  //! Deep copy constructor (can cross spaces)
+  template<typename S, typename O, typename D, typename M, typename ST>
+  CrsMatrix (const std::string& label,
+      const CrsMatrix<S, O, D, M, ST>& mat_)
+  {
+    typename row_map_type::non_const_type rowmap(Kokkos::ViewAllocateWithoutInitializing("rowmap"), mat_.graph.row_map.extent(0));
+    index_type cols(Kokkos::ViewAllocateWithoutInitializing("cols"), mat_.nnz());
+    values = values_type(Kokkos::ViewAllocateWithoutInitializing("values"), mat_.nnz());
+    Kokkos::deep_copy(rowmap, mat_.graph.row_map);
+    Kokkos::deep_copy(cols, mat_.graph.entries);
+    Kokkos::deep_copy(values, mat_.values);
+
+    numCols_ = mat_.numCols();
+    graph = StaticCrsGraphType(cols, rowmap);
+
+#ifdef KOKKOS_USE_CUSPARSE
+    cusparseCreate (&cusparse_handle);
+    cusparseCreateMatDescr (&cusparse_descr);
+#endif // KOKKOS_USE_CUSPARSE
+  }
+
   /// \brief Construct with a graph that will be shared.
   ///
   /// Allocate the values array for subsquent fill.
-  CrsMatrix (const std::string& arg_label,
-             const staticcrsgraph_type& arg_graph) :
-    graph (arg_graph),
-    values (arg_label, arg_graph.entries.extent(0)),
-    numCols_ (maximum_entry (arg_graph) + 1)
+  template<typename DT, typename A1, typename A2, typename A3, typename ST>
```

Comment:
The new mirror constructor above is templated on: `template<typename S, typename O, typename D, typename M, typename ST>` maybe reusing these notation and introducing `L` for layout instead of `A1` would make things easier for maintenance

Reply:
@lucbv Defintely agree, I went through and renamed the args "InOrdinal, InMemTraits, InDevice, ..." and made them consistent for the 3 templated constructors